src/WebJobs.Extensions.DurableTask/Correlation/DurableTaskCorrelationTelemetryInitializer.cs

```diff
@@ -0,0 +1,324 @@
+    /// Original Source is here https://github.com/microsoft/ApplicationInsights-dotnet-server/blob/2.8.0/Src/Common/W3C/W3COperationCorrelationTelemetryInitializer.cs.
+    /// </summary>
+    [Obsolete("Not ready for public consumption.")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+#if DEPENDENCY_COLLECTOR
+    public
+#else
+    internal
+#endif
+        class DurableTaskCorrelationTelemetryInitializer : ITelemetryInitializer
+    {
+        private const string RddDiagnosticSourcePrefix = "rdddsc";
+        private const string SqlRemoteDependencyType = "SQL";
+
+        /// These internal property is copied from W3CConstants
+        /// <summary>Trace-Id tag name.</summary>
+        internal const string TraceIdTag = "w3c_traceId";
+
+        /// <summary>Span-Id tag name.</summary>
+        internal const string SpanIdTag = "w3c_spanId";
+
+        /// <summary>Parent span-Id tag name.</summary>
+        internal const string ParentSpanIdTag = "w3c_parentSpanId";
+
+        /// <summary>Version tag name.</summary>
+        internal const string VersionTag = "w3c_version";
+
+        /// <summary>Sampled tag name.</summary>
+        internal const string SampledTag = "w3c_sampled";
+
+        /// <summary>Tracestate tag name.</summary>
+        internal const string TracestateTag = "w3c_tracestate";
+
+        /// <summary>Default version value.</summary>
+        internal const string DefaultVersion = "00";
+
+        /// <summary>
+        /// Default sampled flag value: may be recorded, not requested.
+        /// </summary>
+        internal const string TraceFlagRecordedAndNotRequested = "02";
+
+        /// <summary>Recorded and requested sampled flag value.</summary>
+        internal const string TraceFlagRecordedAndRequested = "03";
+
+        /// <summary>Requested trace flag.</summary>
+        internal const byte RequestedTraceFlag = 1;
+
+        /// <summary>Legacy root Id tag name.</summary>
+        internal const string LegacyRootIdProperty = "ai_legacyRootId";
+
+        /// <summary>Legacy root Id tag name.</summary>
+        internal const string LegacyRequestIdProperty = "ai_legacyRequestId";
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public DurableTaskCorrelationTelemetryInitializer()
+        {
+            this.ExcludeComponentCorrelationHttpHeadersOnDomains = new HashSet<string>();
+        }
+
+        /// <summary>
+        /// Set of suppress telemetry tracking if you add Host name on this.
+        /// </summary>
+        public HashSet<string> ExcludeComponentCorrelationHttpHeadersOnDomains { get; set; }
+
+        /// <summary>
+        /// Initializes telemetry item.
+        /// </summary>
+        /// <param name="telemetry">Telemetry item.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (this.IsSuppressedTelemetry(telemetry))
+            {
+                this.SuppressTelemetry(telemetry);
```

Comment:
This would suppress the telemetry within the durable task.
Additional work is required to communicate back to the host and suppress the telemetry there:
1. Durable task puts the suppression flag in the payload before returning back to the Functions host.
2. Once Functions host received the suppression flag, respect the flag and stop emitting telemetry events (context propagation is still required).

Reply:
Hi @reyang 
Thank you for the notice! Very good point. 

This issue might be related. 
https://github.com/Azure/azure-functions-durable-extension/issues/593

Maybe this TelemetryInitializer might be  the one fix on the functions host side. 
https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs